scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GoalBuilder.java

```diff
@@ -0,0 +1,104 @@
+package gov.nasa.jpl.aerie.scheduler.server.remotes.postgres;
+
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.scheduler.ActivityCreationTemplate;
+import gov.nasa.jpl.aerie.scheduler.ActivityType;
+import gov.nasa.jpl.aerie.scheduler.CompositeAndGoal;
+import gov.nasa.jpl.aerie.scheduler.Goal;
+import gov.nasa.jpl.aerie.scheduler.OptionGoal;
+import gov.nasa.jpl.aerie.scheduler.PlanningHorizon;
+import gov.nasa.jpl.aerie.scheduler.RecurrenceGoal;
+import gov.nasa.jpl.aerie.scheduler.Time;
+import gov.nasa.jpl.aerie.scheduler.server.models.GoalId;
+import gov.nasa.jpl.aerie.scheduler.server.models.GoalRecord;
+import gov.nasa.jpl.aerie.scheduler.server.models.PlanId;
+import gov.nasa.jpl.aerie.scheduler.server.models.SchedulingDSL;
+import gov.nasa.jpl.aerie.scheduler.server.models.Timestamp;
+import gov.nasa.jpl.aerie.scheduler.server.services.SchedulingDSLCompilationService;
+
+import java.io.IOException;
+
+public class GoalBuilder {
+  static GoalRecord buildGoalRecord(
+      final PlanId planId,
+      final PostgresGoalRecord pgGoal,
+      final Timestamp horizonStartTimestamp,
+      final Timestamp horizonEndTimestamp,
+      final SchedulingDSLCompilationService schedulingDSLCompilationService)
+  {
+    final SchedulingDSL.GoalSpecifier goalSpecifier;
+    try {
+      goalSpecifier = schedulingDSLCompilationService.compileSchedulingGoalDSL(
+          planId,
+          pgGoal.definition(),
+          "goals don't have names?");
+    } catch (SchedulingDSLCompilationService.SchedulingDSLCompilationException | IOException e) {
+      e.printStackTrace();
+      throw new Error("");
```

Comment:
Even if this is an unlikely exception I'd suggest putting some description within the `Error` constructor.